File: favorite-quote-backend/src/main/java/com/david/favoritequote/Service/CategoryService.java
package com.david.favoritequote.Service;

import com.david.favoritequote.Entity.Category;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface CategoryService {
    List<Category> getAllCategory();

    Category getCategoryByID(int categoryID);

    ResponseEntity<Category> addCategory(Category category);

    ResponseEntity<Category> updateCategory(int categoryID, Category category);

    String deleteCategory(int categoryID);
}

File: favorite-quote/src/services/category-service.js
import axios from "axios";

const QUOTE_CATEGORIES_BASE_REST_API_URL =
  "http://localhost:9090/api/categories/";

class CategoryService {
  getAllCategory() {
    return axios.get(QUOTE_CATEGORIES_BASE_REST_API_URL);
  }

  addCategory() {
    //return axios.post(QUOTE_CATEGORIES_BASE_REST_API_URL + "/category" + categoryID , quote)
  }
}

export default new CategoryService();
